doku.php

```diff
@@ -11,6 +11,16 @@
 // update message version - always use a string to avoid localized floats!
 $updateVersion = "50";
 
+// let people know if they are running an unsupported version of PHP
+$required_php = "5.6";
+
+if (version_compare(phpversion(), $required_php, '<')) {
+
+  die('<h2 style="margin:1.2em;text-align:center;color:red;line-height:1.6;" >Web Server Setup Error: The web server currently runs PHP/'.phpversion().'.<br>
+       <span style="color:#606060;" >DokuWiki requires PHP/'.$required_php.' or higher.</span></h3>');
+} 
```

Comment:
I agree with @michitux, this duplicates the `nice_die()` method

Reply:
Sorry for the answer in german! Ich know that's not usually here! And I know the reason why...

Ja, das  mit dem duplicate ist wohl richtig, aber ist das in irgendeiner Hinsicht ein Problem oder ist damit ein ernsthafter Nachteil verbunden? Sollte (wie michitux erwähnt hatte) ein Risiko bestehen, dass bei der Verwendung von nice_die() ältere PHP-Versionen nicht damit arbeiten können, dann würde ja gerade in diesem Fall das Abfangen des Fehlers scheitern. An dieser Stelle sage ich: safety first! Ich halte es also grundsätzlich für am besten, an dieser Stelle so wenig wie irgendmöglich zu tun, damit das auch bis in alle Ewigkeit funktioniert. Und wer weiß, was in Zukunft noch alles in nice_die() hineinprogrammiert wird.

Außerdem gefällt mir nicht, dass in nice_die() ein "DokuWiki Setup Error" ausgegeben wird. So eine Meldung ist für Änfänger irreführend, da es den Eindruck erweckt, der Fehler wäre im DokuWiki zu suchen.

Reply:
> is checking this on every page load again and again, worth it?
Was aus meiner Sicht dafür spricht ist folgende Geschichte: Vor einigen Jahren, als blutiger Anfänger mit DokuWiki und PHP (ich wusste kaum was das ist) funktionierte meine erste DW-Installation 3 Tage lang ganz prima. Bis dann plötzlich ein fataler Fehler auftrat, ganz ähnlich wie dem oben beschriebenen. Ursache: Weil ich das "Versäumnis" begangen hatte, die PHP-Version explizit festzulegen, wurde sie providerseitig nach 3 Tagen auf eine ältere Version zurückgeschaltet. Man kann sich vorstellen, wie viel Kopfzerbrechen mich das gekostet hat, bis ich drauf gekommen bin. Wer rechnet denn mit sowas?

Ich meine, man sollte sehr bedenken, dass das hier ein Problemfeld ist, was alle Nicht-Nerds in schwerste Bedrängnisse bringt. Gerade dann, wenn man den Otto-Normal-User so vehement auffordert mal so eben auf die neue Release zu upgraden.

Reply:
> Could you please keep the discussion in English?. It make no sense to have a discussion people can not follow and chime in.

Sorry, mir ist das Problem absolut bewusst, aber auf Englisch zu schreiben ist für so dermaßen anstrengend und mühsam, dass es für mich nur zwei Alternativen gibt: Entweder mache ich meine (längeren) Ausführungen auf deutsch, oder ich muss es ganz sein lassen, hier mitzuarbeiten. Ich überlasse Dir die Wahl, ich werde mich daran halten.

Reply:
> we may want to use newer PHP features in doku.php some time, which would break this feature again

Das verstehe ich nicht. Warum sollten irgendwelche zukünftigen Änderungen den PHP-Versionscheck kaputt machen, da unser Fehlercheck doch ganz am Anfang stattfindet?

Reply:
Modern PHP in the same file will break the check, because the file is parsed before executed. When the file contains code that is not understood by the used PHP, the parser will fail before any code is executed.

Reply:
> Modern PHP in the same file will break the check, because the file is parsed before executed. When the file contains code that is not understood by the used PHP, the parser will fail before any code (including our check) is executed.

Ok, verstanden. Kann ich eigentlich auch selbst meine eigenen Kommentare als "off topic" markieren?

Reply:
>  [...] the parser will fail before any code (including our check) is executed.

Ok, dann müssen wir in doku.php nur noch den Versionscheck machen und eine PHP-Routine aufrufen, in der alles das getan wird, was jetzt in doku.php getan wird.